mirdata/datasets/slakh.py

```diff
@@ -0,0 +1,408 @@
+            'omitted' tracks are part of slack2100-redux which were found to be
+            duplicates in the original slackh2011.
+        uuid (str): File name of the original MIDI file from Lakh, sans extension
+        lakh_midi_dir (str): Path to the original MIDI file from a fresh download of Lakh
+        normalized (bool): whether the mix and stems were normalized according to the ITU-R BS.1770-4 spec
+        overall_gain (float): gain applied to every stem to make sure mixture does not clip when stems are summed
+
+    Cached Properties:
+        midi (PrettyMIDI): midi data used to generate the mixture audio
+        notes (NoteData): note representation of the midi data
+        multif0 (MultiF0Data): multif0 representation of the midi data
+
+    """
+
+    def __init__(
+        self, mtrack_id, data_home, dataset_name, index, track_class, metadata
+    ):
+        super().__init__(
+            mtrack_id=mtrack_id,
+            data_home=data_home,
+            dataset_name=dataset_name,
+            index=index,
+            track_class=track_class,
+            metadata=metadata,
+        )
+        self.mix_path = self.get_path("mix")
+        self.midi_path = self.get_path("midi")
+        self.metadata_path = self.get_path("metadata")
+
+        # split (train/validation/test) is determined by the relative filepath in the index
+        self.data_split = None  # for baby_slakh, there are no data splits - set to None
+        if index["version"] == "2100-redux":
+            self.data_split = self._multitrack_paths["mix"][0].split(os.sep)[1]
+            assert self.data_split in SPLITS, "{} not in SPLITS".format(self.data_split)
+
+    @property
+    def track_audio_property(self) -> str:
+        return "audio"
+
+    @core.cached_property
+    def _multitrack_metadata(self) -> dict:
+        with open(self.metadata_path, "r") as fhandle:
+            metadata = yaml.safe_load(fhandle)
+        return metadata
+
+    @property
+    def uuid(self) -> Optional[str]:
+        return self._multitrack_metadata.get("UUID")
+
+    @property
+    def lakh_midi_dir(self) -> Optional[str]:
+        return self._multitrack_metadata.get("lmd_midi_dir")
+
+    @property
+    def normalized(self) -> Optional[bool]:
+        return self._multitrack_metadata.get("normalized")
+
+    @property
+    def overall_gain(self) -> Optional[float]:
+        return self._multitrack_metadata.get("overall_gain")
+
+    @core.cached_property
+    def midi(self) -> Optional[pretty_midi.PrettyMIDI]:
+        return io.load_midi(self.midi_path)
+
+    @core.cached_property
+    def notes(self) -> Optional[annotations.NoteData]:
+        return io.load_notes_from_midi(self.midi_path, self.midi)
+
+    @core.cached_property
+    def multif0(self) -> Optional[annotations.MultiF0Data]:
+        return io.load_multif0_from_midi(
+            self.midi_path, self.midi, skip_drums=True, pitch_bend=False
```

Comment:
I believe some (if not all) of the Kontakt patches used to make slakh honor pitch bend events. There are definitely guitar pitch bends in slakh.

Reply:
hmmm is there a way to know for which ones the pitch bends are honors and the ones where it's not? I spot checked a few random ones and they weren't in the audio.

Reply:
(For now i'm going to make a comment and open an issue, and leave it for the future)

Reply:
Unfortunately there's no way to know from looking only at the MIDI. Though I think we could get a rough idea of which patches honor pitch bends by finding a few examples of each patch and inspecting the audio by hand. But to be 100% certain, we'd probably need a way to systematically check _all_ of the pitch bend instances using Label Studio or something similar 😫.